### PR TITLE
Refactor the API and Network libs to remove application logic

### PR DESCRIPTION
**Pull request by tgolen**
### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/145178

### Tests
1. Need to do general regression tests on just about all the functionality you can imagine. The best way to test expired authTokens is to sign in, leave the app for 2+ hours without doing anything, come back and try to do something like post a comment. It should work flawlessly without the flow being interrupted. 
1. Be sure to test going offline and reconnecting

### CODE REVIEW

**Review by cead22 — COMMENTED** [review]
LGTM so far, out of curiosity
- Why the change from API.get to expensifyAPI.get?
- Why the change from `addReportComment` to `expensifyAPI.Report.addComment`? I ask because this adds a third and new way of naming these API commands and I feel we should try to standardize on one. Eg, on web we have things like `Report_AddComment`

**Review by marcaaron — COMMENTED** [review]
Looks great so far! Love the `PusherConnectionManager` idea and feels like there's much stronger separation of concerns now. Have a few initial things but will wait to give a more substantial review.

I agree with @cead22 about the API naming. Personally, I think it would be best if we enforced that all `API` commands matched the commands that we use in `api.php`. It makes the link between these two things more clear which I think can help when you're trying to follow the path a request takes.

You will see `API.Report_AddComment()` and there is no need to see what `command` this uses since the names are the same.

**Review by AndrewGable — DISMISSED** [review]
Tests pass on iOS/Android simulators. Big fan of the refactoring 👍

**Review by cead22 — DISMISSED** [review]

**Review by TomatoToaster — DISMISSED** [review]
It's mostly looking good to me. I had some NAB type comments and I think the comments so far have been addressed.

**Review by marcaaron — CHANGES_REQUESTED** [review]
Really nice changes!! I think we're almost there? Just a heads up I might have some delay in getting around to actually testing this as my work laptop are dead.

**Review by tgolen — COMMENTED** [review]
Thanks for the review, Marc! I have addressed everything and updated the branch.

**Review by marcaaron — DISMISSED** [review]
Thanks for the changes. Looks really great so I'm going to approve. @TomatoToaster would you be up for helping out with testing this while I get my dev environment back up and running?

**Review by TomatoToaster — APPROVED** [review]
LGTM - I'm testing this locally right now.

**Review by cead22 — APPROVED** [review]

**Review by marcaaron — APPROVED** [review]

File: src/libs/API.js [hunk 1]
```diff
@@ -346,148 +196,175 @@ function getAuthToken() {
...
-        partnerPassword: CONFIG.EXPENSIFY.PARTNER_PASSWORD,
+function Authenticate(parameters) {
+    const commandName = 'Authenticate';
+
+    requireParameters([
+        'partnerName',
+        'partnerPassword',
+        'partnerUserID',
+        'partnerUserSecret',
+    ], parameters, commandName);
```
Comment (Pujan92) [inline src/libs/API.js:215]: I think we should handle the thrown error by `requireParameters` here to redirect a user to login page. Issue https://github.com/Expensify/App/issues/58384

**Pull request merged**